src/ops/leaf.rs

```diff
@@ -78,7 +78,9 @@ pub enum LeafOp {
 
 impl Default for LeafOp {
     fn default() -> Self {
-        Self::Noop { ty: Type::Qubit }
+        Self::Noop {
+            ty: Type::graph(AbstractSignature::new_linear(vec![])),
```

Comment:
this should maybe be Type::unit()?

Reply:
Yeah. This shouldn't normally be used but `unit` is a good default.